wandb/sdk/wandb_verify.py

```diff
@@ -0,0 +1,495 @@
+"""
+Utilities for wandb verify
+"""
+
+import getpass
+import os
+import time
+
+import click
+from gql import gql  # type: ignore
+from packaging import version  # type: ignore
+import requests
+import wandb
+
+
+if wandb.TYPE_CHECKING:  # type: ignore
+    from typing import (
+        Any,
+        Dict,
+        List,
+        Optional,
+        Tuple,
+        Union,
+    )
+
+    from wandb.sdk.wandb_artifacts import Artifact
+    from wandb.apis.public import Artifact as ArtifactAPI
+
+    from wandb.apis.internal import Api
+
+PROJECT_NAME = "verify"
+CHECKMARK = u"\u2705"
+RED_X = u"\u274C"
+WARNING_SIGN = u"\u26A0"
+
+
+def print_results(
+    failed_test_or_tests: Optional[Union[str, List[str]]], warning: bool
+) -> None:
+    if warning:
+        color = "yellow"
+        sign = WARNING_SIGN
+    else:
+        color = "red"
+        sign = RED_X
+    if isinstance(failed_test_or_tests, str):
+        print(sign)
+        print(click.style(failed_test_or_tests, fg=color, bold=True))
+    elif isinstance(failed_test_or_tests, list) and len(failed_test_or_tests) > 0:
+        print(sign)
+        print(
+            "\n".join(
+                [click.style(f, fg=color, bold=True) for f in failed_test_or_tests]
+            )
+        )
+    else:
+        print(CHECKMARK)
+
+
+def check_host(host: str) -> bool:
+    if host == "api.wandb.ai":
+        print_results("Cannot run wandb verify against api.wandb.ai", False)
+        return False
+    return True
+
+
+def check_logged_in(api: Api) -> bool:
```

Comment:
This is called in cli.py(line 1591 right now) as the first test. It doesn't indicate a test because it only outputs if you're running against wandb.ai.